README.md

```diff
@@ -1,27 +1,508 @@
+render method:
+
+```js
+    render () {
+        return (
+            <div>
+                {
+                    this.state.posts.map((post) =>
+                        <BlogPost title={post.title} body={post.body} />
+                    )
+                }
+            </div>
+        )
+    }
+```
+
+Keep in mind that we didn't change the `BlogPost` component *at all* — ideally
+you should try to keep the majority of your components "stateless".
+
+At this point, you might wonder why we had to use `setState`, if all `setState`
+does is... well... setting `this.state`. There is a very simple reason for that:
+
+React needs to "know" that a state change happened, since whenever a component's
+"data" (state or props) changes, React has to re-render the component (runs its
+render function to figure out "what changed").
+
+### State vs Props
+
+At this point you might wonder when to use `props` and when to use `state`, but
+there is a good rule of thumb when laying out your application's architecture:
+
+* Use `state` for adding interactivity to your component:
+
+    A dropdown menu for example can be in an `expanded` or `collapsed` state, a
+    modal dialog can be `open` or `closed`.
+
+* Move your state "up":
+
+    In our blog example, the `BlogPost` component could also "manage" its own
+    state by doing a separate XHR request. Nevertheless, this is not an elegant
+    solution, since the blog needs to "know" about the blog posts anyways (how
+    else would it "decide" how many `<BlogPost />` components should be
+    rendered in the first place?).
+
+    Moving **application state** higher up generally speaking simplifies the
+    architecture of your application and allows you to keep the majority of your
+    components stateless.
+
+    State is hard to manage and difficult to test. You should always try to keep
+    your components as pure as possible.
+
+* Start with props:
+
+    When creating the layout of your project, you typically start with a static
+    version and then add interactivity once you reach a certain point.
+
+    During this initial phase, there is almost never a need for state.
+    Everything can be done using props during those initial minutes. Try to keep
+    your components as encapsulated and isolated as possible while you're
+    gradually adding interactivity to individual parts of your application.
+
+    For our `<Blog />` example, we first hardcoded the `<BlogPost />`
+    component's props and then added an API call that updated the `Blog`
+    component's state.
+
+In general, it's usually a good idea to start with a rather static version of
+your application, in which all data is being rendered by gradually passing down
+`props` to child components. Then — as a next step — you can add custom user
+interactions to your app, e.g. what happens when a user presses this button?
+Should it open a dialog by doing `this.setState({dialogOpen: true})`?
+
+## `setState` in detail
```

Comment:
I really like the modal example — it's a real life component that students will immediately be able to relate to! 🎉